services/QuillLMS/app/services/independent_practice_packs_assigner.rb

```diff
@@ -68,10 +68,19 @@ def run
     ActiveRecord::Base.transaction do
       selections
         .map { |selection| UnitTemplate.find(selection['id']) }
-        .each { |ut| Unit.unscoped.find_or_create_by!(unit_template: ut, name: ut.name, user:) }
+        .each { |unit_template| find_or_create_activity_pack(unit_template) }
     end
   end
 
+  private def find_or_create_activity_pack(unit_template)
+    name = unit_template.name
+    return if Unit.unscoped.exists?(unit_template:, name:, user:)
```

Comment:
If there's an existing record, will this actually result in `nil` being returned instead of the existing record?

Reply:
But, the naming is confusing since it's different than the regular find_or_create so I will give it a better name.